runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/LyftProcessJobBundleFactory.java

```diff
@@ -0,0 +1,147 @@
+        Preconditions.checkNotNull(JOB_INFO.get(), "jobInfo is null"),
+        controlServiceServer,
+        loggingServiceServer,
+        retrievalServiceServer,
+        provisioningServiceServer,
+        idGenerator,
+        clientSource);
+  }
+
+  private static class LyftPythonEnvironmentFactory implements EnvironmentFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(LyftPythonEnvironmentFactory.class);
+
+    private final JobInfo jobInfo;
+    private final ProcessManager processManager;
+    private final GrpcFnServer<FnApiControlClientPoolService> controlServiceServer;
+    private final GrpcFnServer<GrpcLoggingService> loggingServiceServer;
+    private final GrpcFnServer<ArtifactRetrievalService> retrievalServiceServer;
+    private final GrpcFnServer<StaticGrpcProvisionService> provisioningServiceServer;
+    private final IdGenerator idGenerator;
+    private final ControlClientPool.Source clientSource;
+
+    private LyftPythonEnvironmentFactory(
+        JobInfo jobInfo,
+        GrpcFnServer<FnApiControlClientPoolService> controlServiceServer,
+        GrpcFnServer<GrpcLoggingService> loggingServiceServer,
+        GrpcFnServer<ArtifactRetrievalService> retrievalServiceServer,
+        GrpcFnServer<StaticGrpcProvisionService> provisioningServiceServer,
+        IdGenerator idGenerator,
+        ControlClientPool.Source clientSource) {
+      this.jobInfo = jobInfo;
+      this.processManager = ProcessManager.create();
+      this.controlServiceServer = controlServiceServer;
+      this.loggingServiceServer = loggingServiceServer;
+      this.retrievalServiceServer = retrievalServiceServer;
+      this.provisioningServiceServer = provisioningServiceServer;
+      this.idGenerator = idGenerator;
+      this.clientSource = clientSource;
+    }
+
+    /** Creates a new, active {@link RemoteEnvironment} backed by a forked process. */
+    @Override
+    public RemoteEnvironment createEnvironment(RunnerApi.Environment environment) throws Exception {
+      String workerId = idGenerator.getId();
+
+      String pipelineOptionsJson = JsonFormat.printer().print(jobInfo.pipelineOptions());
+      HashMap<String, String> env = new HashMap<>();
+      env.put("WORKER_ID", workerId);
+      env.put("PIPELINE_OPTIONS", pipelineOptionsJson);
+      env.put(
+          "LOGGING_API_SERVICE_DESCRIPTOR",
+          loggingServiceServer.getApiServiceDescriptor().toString());
+      env.put(
+          "CONTROL_API_SERVICE_DESCRIPTOR",
+          controlServiceServer.getApiServiceDescriptor().toString());
+      env.put("SEMI_PERSISTENT_DIRECTORY", "/tmp");
+
+      String sdkHarnessEntrypoint = "apache_beam.runners.worker.sdk_worker_main";
+      String executable = "bash";
+      //List<String> args = ImmutableList.of("-c", String.format("'env; python -m %s'", sdkHarnessEntrypoint));
+      List<String> args =
+          ImmutableList.of("-c", String.format("env; python -m %s", sdkHarnessEntrypoint));
```

Comment:
is this HarnessEntryPoint Lyft specific, right? this will be a python module which will be provided by the application and will get passed here. So this JobManager will basically execute that python module.

Reply:
The entry point currently is the standard entry point, nothing Lyft specific. This works for example when the JVM executes within the virtualenv that you would otherwise also use for development. That's how I run the job server locally.

Reply:
Another quick question, it is also making an assumption that the arguments will be passed through the environment variable. is it a correct assumption?

Reply:
Do you mean `env` in the command? That's just for debugging.

The Python SDK harness can be invoked using the environment variables, that's how it is done in the Docker container also: https://github.com/apache/beam/blob/release-2.7.0/sdks/python/container/boot.go#L110